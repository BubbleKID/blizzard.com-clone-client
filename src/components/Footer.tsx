import React from "react";

interface LinkProps {
    name: string;
    url: string;
    fontSize: number;
}

const Link = (props: LinkProps) => {
    return (
        <React.Fragment >
            <a style={{textTransform: "uppercase", fontSize: props.fontSize}} href={props.url}>{props.name}</a>
        </React.Fragment>
    );
};

export default function Footer() {
    const links = [
        { name: 'Careers', url: '#'},
        { name: 'About', url: '#'},
        { name: 'Support', url: '#'},
        { name: 'Contact Us', url: '#'},
        { name: 'Press', url: '#'},
        { name: 'API', url: '#'},
        { name: 'Site Map', url: '#'}
    ];

    const smallLinks = [
        { name: 'Privacy', url: '#'},
        { name: 'Legal', url: '#'},
        { name: 'Terms', url: '#'},
        { name: 'Cookie Policy', url: '#'},
        { name: 'Cookie Setting', url: '#'}
    ];

    return (
        <div>
            {links.map(link => (
                <Link key={link.name} name={link.name} url={link.url} fontSize={14}></Link>
            ))}
            <p>All games, one app:Battle.net Desktop App</p>
            <p>Â©2021 BLIZZARD ENTERTAINMENT, INC. ALL RIGHTS RESERVED.</p>
            <p>All trademarks referenced herein are the properties of their respective owners.</p>
            {smallLinks.map(link => (
                <Link key={link.name} name={link.name} url={link.url} fontSize={12}></Link>
            ))}
        </div>
    );
}