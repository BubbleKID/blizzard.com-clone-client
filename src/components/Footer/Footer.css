.links {
    color: rgba(255,255,255,0.7);
    text-transform: uppercase;
    text-decoration-line: none;
}

.links:nth-child(n+2):before {
    content: '';
    display: inline-block;
    height: 10px;
    width: 1px;
    background-color: rgba(255,255,255,0.3);
    margin: 0 15px;
    pointer-events: none;
}

p {
    color: rgba(255,255,255,0.7);
}

a {
    text-decoration-line: none;
    color: rgba(255,255,255,0.7);
    transition: color .2s;
}

a:hover {
    color: #fff;
}

.footer-container {
    text-align: center;
}

.navbar-icon {
    height: 1em;
    width: 1.5em;
}

.navbar-icon path {
    fill: #00aeff;
}

.navbar-app {
    cursor: pointer;
    color: #00aeff;
}