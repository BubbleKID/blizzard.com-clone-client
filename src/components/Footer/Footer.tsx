import React from "react";
import LanguageSelector from './../LanguageSelector';
import './Footer.css';
import { ReactComponent as BNetIcon } from '../../images/Blizzard Entertainment-16.svg';

interface LinkProps {
    name: string;
    url: string;
    fontSize: number;
}

const Link = (props: LinkProps) => {
    return (
        <React.Fragment >
            <a style={{fontSize: props.fontSize}} className="links" href={props.url}>{props.name}</a>
        </React.Fragment>
    );
};

export default function Footer() {
    const links = [
        { name: 'Careers', url: '#'},
        { name: 'About', url: '#'},
        { name: 'Support', url: '#'},
        { name: 'Contact Us', url: '#'},
        { name: 'Press', url: '#'},
        { name: 'API', url: '#'},
        { name: 'Site Map', url: '#'}
    ];

    const smallLinks = [
        { name: 'Privacy', url: '#'},
        { name: 'Legal', url: '#'},
        { name: 'Terms', url: '#'},
        { name: 'Cookie Policy', url: '#'},
        { name: 'Cookie Setting', url: '#'}
    ];

    return (
        <div className="footer-container"  style={{margin: '100px 0'}}>
            <div className="languague"></div>
            <div className="logo"></div>
            <div>
                {links.map(link => (
                    <Link key={link.name} name={link.name} url={link.url} fontSize={17}></Link>
                ))}
            </div>
            <div style={{fontSize: '16px', margin: '40px 0'}}>
                <p>All games, one app: <BNetIcon className="navbar-icon"/><a className="navbar-app">Battle.net Desktop App</a></p>
            </div>
            <p style={{fontSize: '13px', margin: '0'}}>Â©2021 BLIZZARD ENTERTAINMENT, INC. ALL RIGHTS RESERVED.</p>
            <p style={{fontSize: '13px', margin: '0'}}>All <a href="#" style={{fontWeight: 'bold'}}>trademarks</a> referenced herein are the properties of their respective owners.</p>
            <div>
                {smallLinks.map(link => (
                    <Link key={link.name} name={link.name} url={link.url} fontSize={13}></Link>
                ))}
            </div>
        </div>
    );
}