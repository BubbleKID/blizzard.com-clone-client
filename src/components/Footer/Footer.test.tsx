import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Footer from './Footer';

test('Footer should be rendered', () => {
    const component = TestRenderer.create(<Footer/>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});