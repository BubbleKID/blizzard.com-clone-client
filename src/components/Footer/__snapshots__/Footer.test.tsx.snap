// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should be rendered 1`] = `
<div
  className="footer-container"
  style={
    Object {
      "margin": "100px 0",
    }
  }
>
  <div
    className="languague"
  />
  <div
    className="logo"
  />
  <div>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Careers
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      About
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Support
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Contact Us
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Press
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      API
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Site Map
    </a>
  </div>
  <div
    style={
      Object {
        "fontSize": "16px",
        "margin": "40px 0",
      }
    }
  >
    <p>
      All games, one app: 
      <ReactComponent
        className="navbar-icon"
      />
      <a
        className="navbar-app"
      >
        Battle.net Desktop App
      </a>
    </p>
  </div>
  <p
    style={
      Object {
        "fontSize": "13px",
        "margin": "0",
      }
    }
  >
    Â©2021 BLIZZARD ENTERTAINMENT, INC. ALL RIGHTS RESERVED.
  </p>
  <p
    style={
      Object {
        "fontSize": "13px",
        "margin": "0",
      }
    }
  >
    All 
    <a
      href="#"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      trademarks
    </a>
     referenced herein are the properties of their respective owners.
  </p>
  <div>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Privacy
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Legal
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Terms
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Cookie Policy
    </a>
    <a
      className="links"
      href="#"
      style={
        Object {
          "fontSize": 13,
        }
      }
    >
      Cookie Setting
    </a>
  </div>
</div>
`;
